Source/NSCollectionView.m

```diff
@@ -1114,4 +1456,558 @@ - (BOOL) wantsPeriodicDraggingUpdates
   return YES;
 }
 
+/* New methods for later versions of macOS */
+
+// 10.11 methods...
+
+/* Locating Items and Views */
+
+- (NSArray *) visibleItems
+{
+  return _visibleItems;
+}
+
+- (NSSet *) indexPathsForVisibleItems
+{
+  return _indexPathsForVisibleItems;
```

Comment:
Again you seem to be keeping two variables with connected content. It would be better to just store one of these and compute the other on the fly. The methods `indexPathForItem:` and `itemAtIndexPath` would be helpful here.

Reply:
Most likely you closed this in error, mistaking it for the comment on line 526.

Reply:
```
- (NSSet *) indexPathsForVisibleItems
{
  NSMutableSet *result = [NSMutableSet setWithCapacity: [_visibleItems count]];

  FOR_IN(NSCollectionViewItem*, item, _visibleItems)
    {
      NSIndexPath *p = [_itemsToIndexPaths objectForKey: item];
      [result addObject: p];
    }
  END_FOR_IN(_visibleItems);
  
  return [NSSet setWithSet: result]; // return immutable copy
}
```
This is what is currently at line 1473-1485 in NSCollectionView.m

Reply:
I'm resolving this...  it has been addressed.